shared/src/main/scala/com/cibo/evilplot/plot/BoxPlot.scala

```diff
@@ -98,6 +125,67 @@ object BoxPlot {
   )(implicit theme: Theme): Plot = {
     val summaries =
       data.map(dist => if (dist.nonEmpty) Some(BoxPlotSummaryStatistics(dist, quantiles)) else None)
+    custom(summaries, boxRenderer, pointRenderer, quantiles, spacing, None, boundBuffer)
+  }
+
+  /** Create clustered box plots for a sequence of distributions.
+    *
+    * @param data the clusters of distributions to plot
+    * @param boxRenderer the `BoxRenderer` to use to display each distribution
+    * @param pointRenderer the `PointRenderer` used to display outliers
+    * @param quantiles quantiles to use for summary statistics.
+    *                  defaults to 1st, 2nd, 3rd quartiles.
+    * @param spacing spacing how much spacing to put between boxes
+    * @param clusterSpacing ho much spacing to put between clusters of boxes
+    * @param boundBuffer expand bounds by this factor
+    */
+  def clustered(
+    data: Seq[Seq[Seq[Double]]],
+    quantiles: (Double, Double, Double) = (0.25, 0.50, 0.75),
+    spacing: Option[Double] = None,
+    clusterSpacing: Option[Double] = None,
+    boundBuffer: Option[Double] = None,
+    boxRenderer: Option[BoxRenderer] = None,
+    pointRenderer: Option[PointRenderer] = None
+  )(implicit theme: Theme): Plot = {
+    println(data.length)
+    val summaries = data.zipWithIndex.flatMap {
+      case (cluster, clusterIndex) =>
+        cluster.map(
+          dist =>
+            if (dist.nonEmpty) Some(BoxPlotSummaryStatistics(dist, quantiles, cluster = clusterIndex))
+            else None)
+    }
+    custom(
+      summaries,
+      boxRenderer,
+      pointRenderer,
+      quantiles,
+      spacing,
+      Some(clusterSpacing.getOrElse(theme.elements.clusterSpacing)),
+      boundBuffer)
+  }
+
+  /** Create custom box plots for a sequence of distributions.
+    *
+    * @param data the distributions to plot
+    * @param boxRenderer the `BoxRenderer` to use to display each distribution
+    * @param pointRenderer the `PointRenderer` used to display outliers
+    * @param quantiles quantiles to use for summary statistics.
+    *                  defaults to 1st, 2nd, 3rd quartiles.
+    * @param spacing spacing how much spacing to put between boxes
+    * @param boundBuffer expand bounds by this factor
+    */
+  def custom(
+    summaries: Seq[Option[BoxPlotSummaryStatistics]],
```

Comment:
why are we using `Seq[Option[BoxPlotSummaryStatistics]]` here? what happens with `None`s?